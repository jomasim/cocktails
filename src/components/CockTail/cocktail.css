.ant-card-meta-description {
  color: #c3aa8a;
}
.ant-card-meta-title {
  color: #fff;
}
.ant-card.ant-card-bordered.ant-card-hoverable {
  background: #222126;
  border-color: #222126;
  box-shadow: 0 19px 38px rgba(0, 0, 0, 0.3), 0 15px 12px rgba(0, 0, 0, 0.22);
  width: 240px;
}
.more-btn {
  margin-top: 20px;
  background: #d22747;
  color: #fff;
  width: 100%;
  font-size: 16px;
  border-color: #d22747;
  border-radius: 4px;
}
.meta-detail {
  margin-top: 10px;
}
.meta-co {
  display: flex;
}
.meta-item-right {
    color: #fff;
}
.meta-item-left {
    color: #c3aa8a;
}
