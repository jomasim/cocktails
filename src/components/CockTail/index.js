import React from 'react'
import { Card, Button } from 'antd'
import './cocktail.css'

const CockTail = ({ data }) => (
  <Card
    hoverable
    style={{ width: '300px', margin: '10px' }}
    cover={
      <img
        alt='alt'
        src={data.strDrinkThumb}
      />
    }
  >
    <div className='meta-detail'>
      <p style={{ color: '#fff', fontSize: '20px' }}>{data.strDrink}</p>
      <div className='meta-co'>
        <span className='meta-item-left'>Category:</span>
        <span className='meta-item-right'>{data.strCategory}</span>
      </div>
      <div className='meta-co'>
        <span className='meta-item-left'>Glass:</span>
        <span className='meta-item-right'>{data.strGlass}</span>
      </div>
    </div>
    <Button className='more-btn'>More</Button>
  </Card>
)

export default CockTail
