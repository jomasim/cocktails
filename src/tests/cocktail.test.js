import React from 'react'
import CockTail from '../components/CockTail'
import { shallow } from 'enzyme'


const data = {
  idDrink: '178326',
  strDrink: 'White Wine Sangria',
  strCategory: 'Punch / Party Drink',
  strGlass: 'Pitcher'
}

it('should render  cocktail card ', () => {
  const wrapper = shallow(<CockTail data={data} />)
  expect(wrapper.length).toBe(1)
})
it('matches snapshot', ()=>{
  const wrapper = shallow(<CockTail data={data} />)
  expect(wrapper).toMatchSnapshot()
})
