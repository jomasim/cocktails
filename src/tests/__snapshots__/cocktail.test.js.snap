// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Card
  cover={
    <img
      alt="alt"
    />
  }
  hoverable={true}
  style={
    Object {
      "margin": "10px",
      "width": "300px",
    }
  }
>
  <div
    className="meta-detail"
  >
    <p
      style={
        Object {
          "color": "#fff",
          "fontSize": "20px",
        }
      }
    >
      White Wine Sangria
    </p>
    <div
      className="meta-co"
    >
      <span
        className="meta-item-left"
      >
        Category:
      </span>
      <span
        className="meta-item-right"
      >
        Punch / Party Drink
      </span>
    </div>
    <div
      className="meta-co"
    >
      <span
        className="meta-item-left"
      >
        Glass:
      </span>
      <span
        className="meta-item-right"
      >
        Pitcher
      </span>
    </div>
  </div>
  <Button
    className="more-btn"
    onClick={[Function]}
  >
    More
  </Button>
</Card>
`;
